File: FirstAkkaApplication/src/main/scala/akka/first/app/scala/MapReduceApplication.scala
package akka.first.app.scala

import scala.collection.immutable._

import akka.actor.ActorSystem
import akka.actor.Props
import akka.actor.actorRef2Scala
import akka.first.app.scala.actors.MasterActor

class Word(val word:String,val count:Int)
case class Result()
class MapData(val dataList: List[Word])
class ReduceData(val reduceDataMap: Map[String, Int])


object MapReduceApplication {

	def main(args: Array[String]) {
		val _system = ActorSystem("MapReduceApp")
		val master = _system.actorOf(Props[MasterActor], name = "master")

		master ! "The quick brown fox tried to jump over the lazy dog and fell on the dog"
		master ! "Dog is man's best friend"
		master ! "Dog and Fox belong to the same family"

		Thread.sleep(500)
		master ! new Result

		Thread.sleep(500)
		_system.shutdown
    println("Scala done!")
	}
}

File: FirstAkkaApplication/src/main/java/akka/first/app/java/MapReduceApplication.java
package akka.first.app.java;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.first.app.java.actors.MasterActor;
import akka.first.app.java.messages.Result;

public class MapReduceApplication {

	public static void main(String[] args) throws Exception {

		ActorSystem _system = ActorSystem.create("MapReduceApp");
		
		ActorRef master = _system.actorOf(new Props(MasterActor.class),"master");
		
		master.tell("The quick brown fox tried to jump over the lazy dog and fell on the dog");
		master.tell("Dog is man's best friend");
		master.tell("Dog and Fox belong to the same family");
		
		Thread.sleep(500);
		
		master.tell(new Result());
		
		Thread.sleep(500);
		
		_system.shutdown();
	}
}
